service/pixelated/resources/account_recovery_resource.py

```diff
@@ -63,7 +70,12 @@ def success_response(response):
 
         def error_response(failure):
             log.warn(failure)
-            request.setResponseCode(INTERNAL_SERVER_ERROR)
+            if failure.type is InvalidPasswordError or failure.type is EmptyFieldsError:
+                request.setResponseCode(BAD_REQUEST)
+            elif failure.type is UnauthorizedLogin:
+                request.setResponseCode(UNAUTHORIZED)
+            else:
+                request.setResponseCode(INTERNAL_SERVER_ERROR)
```

Comment:
hey @anikarni 
I extracted this response code flow to the common parent `BaseResource` and it should be easier for adding new exceptions. I noticed however that the `LoginResource` doesn't apply these response codes, since its implementation redirects to `/login?auth-error`, overriding any previous code to 302. I think this refactoring is still valuable though. Please take a look.